### PR TITLE
feat(checkout): CUST-1460 Add documentation for the new V3 ValidateCredentials endpoint

### PR DESCRIPTION
**Pull request by valentindellangela**
## Description:
Updating the public facing Customer API docs with the new V3 ValidateCredentials endpoint we created - captured in this ticket https://jira.bigcommerce.com/browse/CUST-1424.

## Testing:
Swagger Editor
image

cc @bigcommerce/dev-docs @bigcommerce/checkout

### CODE REVIEW

File: reference/customers.v3.yml [hunk 1]
```diff
@@ -712,6 +713,62 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidateCustomerCredentialsResponseObject'
+              examples:
+                data:
+                  value:
+                    is_valid: true
+        '429':
+          description: |
+            This endpoint has special rate limiting protections to protect against abuse.
```
Comment (capsula4) [inline reference/customers.v3.yml:748]: `Allowed number of requests exceeded` ?

File: reference/customers.v3.yml [hunk 2]
```diff
@@ -5083,3 +5140,26 @@ components:
...
+      properties:
+        email:
+          type: string
+          description: Customer email address
+        password:
+          type: string
+          description: Customer password
+        channel_id:
+          type: integer
+          description: ChannelId to check the customer credentials against
```
Comment (capsula4) [inline reference/customers.v3.yml:5158]: maybe worth explaining what happens when channel_id is not provided?

File: reference/customers.v3.yml (hunk 1 above)
Comment (valentindellangela) [reply reference/customers.v3.yml:748]: I wasn't sure how to articulate it so I've reused the generic one we have for instance for the actual V2 endpoint.

File: reference/customers.v3.yml [hunk 3]
```diff
@@ -5083,3 +5140,26 @@ components:
...
+          type: string
+          description: Customer password
+        channel_id:
+          type: integer
+          description: ChannelId to check the customer credentials against - If this field is empty we will use channel 1
+    ValidateCustomerCredentialsResponseObject:
+      title: ValidateCustomerCredentialsResponseObject
+      type: object
+      properties:
+        is_valid:
```
Comment (davidchin) [inline reference/customers.v3.yml:5163]: Missing `customer_id`, which get returned if credentials are valid? e.g.:
```
customer_id:
  type: integer
  nullable: true
  example: 1
```

File: reference/customers.v3.yml [hunk 4]
```diff
@@ -712,6 +713,62 @@ paths:
...
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ValidateCustomerCredentialsObject'
+            examples:
+              data:
+                value:
+                  email: 'foo@bar.com'
+                  password: 'password'
+                  channel_id: 1
+      responses:
```
Comment (davidchin) [inline reference/customers.v3.yml:735]: Missing `422` response, which gets returned if the request parameters are incorrect?

File: reference/customers.v3.yml (hunk 3 above)
Comment (valentindellangela) [reply reference/customers.v3.yml:5163]: 🤦  I looked at the wrong screen where it wasn't returned. yeah will update

File: reference/customers.v3.yml [hunk 5]
```diff
@@ -712,6 +713,86 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidateCustomerCredentialsResponseObject'
+              examples:
+                data:
+                  value:
+                    is_valid: true
+                    customer_id: 1
+        '422':
+          description: The *Customer* was not valid. This is the result of missing required fields or trying to edit a read only field. See the response for more details.
```
Comment (davidchin) [inline reference/customers.v3.yml:748]: Maybe just say `This is the result of missing required fields. See the response for more details.`

**Review by davidchin — APPROVED** [review]
LGTM 👍

**Pull request merged**